src/App.tsx

```diff
@@ -58,6 +58,36 @@ function unslugify(raw: string): string {
     .join(' ');
 }
 
+function calculateTeamCompleteness(roundPlayers: (FullPlayerRoundInfo | undefined)[]): {
+  playersPlayed: number;
+  playersAvailable: number;
+  teamIsComplete: boolean;
+} {
+  const playersPlayed = roundPlayers.filter(player => player?.played).length;
+  const playersAvailable = Math.min(11, roundPlayers.length - roundPlayers.filter(player => player?.benched).length);
+  let firstUnplayed: number | undefined = undefined;
+  let lastPlayed: number | undefined = undefined;
+  for (let index = 0; index < roundPlayers.length; index++) {
+    const player = roundPlayers[index];
+    if (index < (firstUnplayed ?? roundPlayers.length) && !(player?.played || player?.benched || player?.injured)) {
```

Comment:
Can the first part of this just check for `firstUnplayed !== undefined` instead?

Reply:
Actually, maybe we can make a lot of this logic more direct by employing .some() and .findIndex/.findLastIndex to get to the values we want more directly and explicitly. I'm having some trouble following all the nuances.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLastIndex

Reply:
Good point – it became much more readable and direct with this approach